urllib3/contrib/pyopenssl.py

```diff
@@ -465,7 +461,7 @@ def ssl_wrap_socket(sock, keyfile=None, certfile=None, cert_reqs=None,
         try:
             cnx.do_handshake()
         except OpenSSL.SSL.WantReadError:
-            rd, _, _ = select.select([sock], [], [], sock.gettimeout())
+            rd = util.wait_for_read(sock, sock.gettimeout())
```

Comment:
Longer term we should re-evaluate this. We should try to avoid the overhead of repeatedly creating selectors if we can.